src/cli.rs

```diff
@@ -121,16 +121,16 @@ pub fn parse_args_from(mut args: Vec<OsString>) -> crate::Result<ParsedArgs> {
             }
 
             // Parse flags
-            let (files, mut flags) = oof::filter_flags(args, &flags_info)?;
+            let (files, flags) = oof::filter_flags(args, &flags_info)?;
             let files = files.into_iter().map(PathBuf::from).collect();
 
-            let output_folder = flags.take_arg("output").map(PathBuf::from);
+            let output_folder = flags.arg("output").map(PathBuf::from);
```

Comment:
@fabricio7p I have now fixed the `--output` error, we were using `take_arg` which removes the flag from our `oof::Flags` struct, that's why we couldn't find that later.

Not, by using `.arg()` instead, we clone (kindof) the content, and now the `output_folder` can be found in our `Command::Decompress` as well as `oof::Flags`.

I adapted the tests, and now they are passing correctly, sorry for the late response.